crates/pink/pink-extension/src/chain_extension.rs

```diff
@@ -59,66 +124,66 @@ pub trait PinkExt {
     /// Values stored in cache can only be read in query functions.
     ///
     /// Alwasy returns `Ok(())` if it is called from a command context.
-    #[ink(extension = 6, handle_status = false, returns_result = false)]
+    #[ink(extension = 6, handle_status = true)]
```

Comment:
When `handle_status = true`, ink framework would check the status code(which is just the return value of the ffi fn) and, if non-zero, convert it to the `Result::Err`.
When `handle_status = false`, it would decode the `T` of `-> Result<T, E>` without checking the status code, no way to pass the value of `E`.

In ink-3.0, we can use `handle_status = false, returns_result = false` to bypass the smart decoding mechanism and just let it decode the entire `Result<T, E>` from the output buffer. But `returns_result = false` was removed in ink-4, and we have other choices.